lib/kubernetes-deploy.rb

```diff
@@ -84,9 +101,6 @@ def run
     wait_for_completion(resources)
 
     report_final_status(resources)
-  rescue FatalDeploymentError => error
-    KubernetesDeploy.logger.fatal(error.message)
-    exit 1
```

Comment:
This went into `with_friendly_errors` which is only used in CLI. In tests, we don't want to `rescue` and exit on exception because we want to test these exceptions.